authors/apps/search/tests/test_search_view.py

```diff
@@ -0,0 +1,130 @@
+from django.contrib.auth import get_user_model
+from django.test import TestCase
+from django.urls import reverse
+from rest_framework import status
+from rest_framework.test import APIClient
+
+from authors.apps.articles.models import Articles
+
+
+def sample_article(user_id):
+    """
+    handle Create a sample article
+    :param user_id:
+    :return: [Article object]
+    """
+    return Articles.objects.create(
+        author_id=user_id,
+        title="the 3 musketeers",
+        body="is a timeless story",
+        description="not written by me",
+        tags=['musk', "three", "teers"]
+    )
+
+
+def sample_user(name):
+    """
+    handle Create a sample user
+    :return: [User object]
+    """
+    user = get_user_model().objects.create_user(
+        f'{name}@gmail.com',
+        f'user_{name}',
+        'T35tP45w0rd'
+    )
+    user.is_verified = True
+    user.save()
+    return user
```

Comment:
Is there a reason the above functions were not included in your test `setUp` method?

Reply:
this is to prevent me from creating multiple users inside the setUp method. like i want user1 to have a username of test1 and user2 to have a username of test2. this will make me create two users inside my setup which is not ideal.

Reply:
i am trying to prevent this:
```
self.user = User.objects.create(
            username='user',
            email='user@mail.com',
            password='password'
        )
        self.user.is_verified = True
        self.user.save()

        self.user2 = User.objects.create(
            username='user2',
            email='user2@mail.com',
            password='password'
        )
        self.user2.is_verified = True
        self.user2.save()
```
instead i did that.

Reply:
The idea behind `setUp` is to provide basic housekeeping before each test is run and so by having the created users in the setup, you don't have to call the `sample_user` function in each and every test as you are doing. Alternatively you can have the function itself under `setUp` and then call it under `setUp` to take advantage of it without repeating yourself. Of note is that, you don't have to change this, it is all up to you in the end(as long as you have a valid reason😅).

Reply:
I have tryed to add a few inside the setUp method to prevent duplication in my test code.